src/gt4py/backend/numpy_backend.py

```diff
@@ -187,7 +187,71 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr):
         return source
 
     def visit_If(self, node: gt_ir.If):
-        raise NotImplementedError("The numpy backend does not support runtime if statements.")
+        sources = []
+        self.conditions_depth += 1
+        sources.append(
+            "__condition_{level} = {condition}".format(
+                level=self.conditions_depth, condition=self.visit(node.condition)
+            )
+        )
+
+        for stmt in node.main_body.stmts:
```

Comment:
Is it possible to merge the two for-loops together since they share so much code?

Reply:
It looks like it's done